### PR TITLE
Ea/hmis data quaility goals

### PR DESCRIPTION
**Pull request by eanders**

### CODE REVIEW

**Comment by eanders** [comment]
@awisema this is ready to review.  There will probably be one more iteration on goals to include charts, but we'll get some feedback on included metrics and do more QA of the numbers first.
Here's the current setup:
HMIS DQ Tool

File: drivers/hmis_data_quality_tool/app/models/hmis_data_quality_tool/dq_concern.rb [hunk 1]
```diff
@@ -36,12 +36,18 @@ def self.calculate_issues(report_items, report)
 
     def self.detail_headers_for(slug)
       section = sections[slug.to_sym]
-      return detail_headers unless section
+      headers = detail_headers.map do |k, v|
+        [
+          k,
+          v.except(:translator),
+        ]
+      end.to_h
```
Comment (awisema) [inline drivers/hmis_data_quality_tool/app/models/hmis_data_quality_tool/dq_concern.rb]:
```suggestion
      headers = detail_headers.transform_values { |v| v.except(:translator) }
```

File: drivers/hmis_data_quality_tool/app/models/hmis_data_quality_tool/enrollment.rb [hunk 2]
```diff
@@ -601,6 +601,7 @@ def self.sections # rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexit
 
             item.lot > 90 && GrdaWarehouse::Hud::Project::RESIDENTIAL_PROJECT_TYPES[:es].include?(item.project_type)
           },
+          es_stay_length: 90,
         },
         lot_es_180_issues: {
           title: 'Possible Missed Exit - ES, Time in Enrollment 180 Days or More',
```
Comment (awisema) [inline drivers/hmis_data_quality_tool/app/models/hmis_data_quality_tool/enrollment.rb:607]: I don't think the title matches description and the limiter for some of these... Is it _180 or more_ (>=180) or _more than 180_ (>180)?

File: drivers/hmis_data_quality_tool/app/models/hmis_data_quality_tool/report.rb [hunk 3]
```diff
@@ -229,99 +245,162 @@ def items_for(key)
...
+
+    private def cache_key
+      [self.class.name, id]
+    end
+
+    def uncache!
+      Rails.cache.delete(cache_key)
+    end
+
+    # FIXME: this should run at the end of the run and cache the results in redis
```
Comment (awisema) [inline drivers/hmis_data_quality_tool/app/models/hmis_data_quality_tool/report.rb:329]:
```suggestion
```

File: drivers/hmis_data_quality_tool/app/models/hmis_data_quality_tool/report.rb [hunk 4]
```diff
@@ -338,10 +417,53 @@ def overall_current_living_situation_count
...
+      @client_ids_per_project ||= enrollments.pluck(:project_id, :destination_client_id).group_by(&:shift)
+      @client_ids_per_project[project.id]&.flatten
+    end
+
+    private def count_category_setup
+      setup = { 'Overall' => { invalid_count: 0, total: 0 } }
+      categories.each do |cat|
+        setup[cat] = { invalid_count: 0, total: 0 }
+      end
+      setup
```
Comment (awisema) [inline drivers/hmis_data_quality_tool/app/models/hmis_data_quality_tool/report.rb]: You added 'Overall' to `categories` above, so:
```suggestion
      {}.tap do |setup|
      categories.each do |cat|
        setup[cat] = { invalid_count: 0, total: 0 }
      end
```

**Review by awisema — APPROVED** [review]
Some minor suggested edits, but looks good!

File: drivers/hmis_data_quality_tool/app/models/hmis_data_quality_tool/report.rb (hunk 4 above)
Comment (eanders) [reply drivers/hmis_data_quality_tool/app/models/hmis_data_quality_tool/report.rb]: Whoops!

File: drivers/hmis_data_quality_tool/app/models/hmis_data_quality_tool/enrollment.rb (hunk 2 above)
Comment (eanders) [reply drivers/hmis_data_quality_tool/app/models/hmis_data_quality_tool/enrollment.rb:607]: Good call.  Made them all match the title >= and days or more.

**Pull request merged**